### PR TITLE
ddl: Support creating anonymous index

### LINKED ISSUE #1785: Add the anonymous index
1. What did you expect to see?

mysql> create table ttt (a int, b int);
Query OK, 0 rows affected (0.02 sec)

mysql> alter table ttt add key (a);
Query OK, 0 rows affected (0.06 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> alter table ttt add key (b);
Query OK, 0 rows affected (0.02 sec)
Records: 0  Duplicates: 0  Warnings: 0

2.What did you see instead?

mysql> create table ttt (a int, b int);
Query OK, 0 rows affected (2.01 sec)

mysql>  alter table ttt add key (a);
Query OK, 0 rows affected (20.00 sec)

mysql>  alter table ttt add key (b);
ERROR 1831 (HY000): [schema:1831]CREATE INDEX: index already exist

### PR DESCRIPTION
**Pull request by zimulala**
Fix #1785

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -1206,6 +1206,11 @@ func (d *ddl) CreateIndex(ctx context.Context, ti ast.Ident, unique bool, indexN
 		return errors.Trace(err)
 	}
 
+	// Deal with anonymous index.
+	if len(indexName.L) == 0 {
+		indexName = idxColNames[0].Column.Name
```
Comment (hanfei1991) [inline ddl/ddl.go]: 1. Will the index(c1,c2) and index(c1) conflict? Add the test.
2. Is it possible that idxColNames is empty?

Comment (zimulala) [inline ddl/ddl.go]: 2. If idxColNames is nil, it will be an error in parser.

**Comment by hanfei1991** [comment]
LGTM

File: ddl/ddl.go (hunk 1 above)
Comment (coocood) [inline ddl/ddl.go]: If the is name conflict, MySQL automatically appends `_2` at the end;
```
create table t (c1 int, c2 int, index (c1,c2), index (c1));
```
The index name is `c1` for `(c1, c2)`, `c1_2` for `(c1)`

Comment (shenli) [inline ddl/ddl.go]: Add a TODO here.

**Comment by zimulala** [comment]
PTAL @hanfei1991 @coocood

**Comment by hanfei1991** [comment]
LGTM

File: ddl/ddl.go [hunk 2]
```diff
@@ -1190,6 +1190,21 @@ func (d *ddl) TruncateTable(ctx context.Context, ti ast.Ident) error {
...
 }
 
+func getAnonymousIndex(t table.Table, colName model.CIStr) model.CIStr {
+	id := 2
+	l := len(t.Indices())
+	indexName := colName
+	for i := 0; i < l; i++ {
+		if t.Indices()[i].Meta().Name.L == indexName.L {
+			indexName.L = fmt.Sprintf("%s_%d", colName.L, id)
+			indexName.O = fmt.Sprintf("%s_%d", colName.O, id)
```
Comment (coocood) [inline ddl/ddl.go]: Use `model.NewCIStr`

File: ddl/ddl.go [hunk 3]
```diff
@@ -1190,6 +1190,21 @@ func (d *ddl) TruncateTable(ctx context.Context, ti ast.Ident) error {
...
+			i = 0
```
Comment (coocood) [inline ddl/ddl.go]: The next loop, `i` value is 1

Comment (hanfei1991) [inline ddl/ddl.go]: i value should be id - 1

**Comment by coocood** [comment]
LGTM

**Pull request merged**